server/mdm/nanomdm/storage/mysql/mysql.go

```diff
@@ -67,8 +73,20 @@ func WithDeleteCommands() Option {
 	}
 }
 
+func WithAsyncLastSeen(cap int, interval time.Duration) Option {
+	return func(c *config) {
+		c.asyncCap = cap
+		c.asyncInterval = interval
+	}
+}
+
 func New(opts ...Option) (*MySQLStorage, error) {
-	cfg := &config{logger: log.NopLogger, driver: "mysql"}
+	const (
+		asyncLastSeenFlushInterval = 2 * time.Second
```

Comment:
What happens when the same host is seen multiple times during this interval?

Reply:
The host ID is added to a set so it's always deduplicated (and sorted to help prevent deadlocks) when processed: https://github.com/fleetdm/fleet/blob/b8718bc0f9c9075ec007ced7438a345e2c3340ef/server/mdm/nanomdm/storage/mysql/async.go#L65